File: Python3/프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = [0] * len(targets)
    keyorder = 101
    for i, target in enumerate(targets):
        for t in target:
            keyorder = 101
            for km in keymap:
                if t in km:
                    keyorder = min(keyorder, km.index(t))
            if keyorder == 101:
                answer[i] = -1
                break
            answer[i] += keyorder + 1
        
    return answer

File: C++/프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer(targets.size());
    vector<int> indexes(27, 101);
    
    for(auto& key : keymap) {
        for(int i = 'A'; i <= 'Z'; i++) {
            for(int k = 0; k < key.size(); k++) {
                if(i == key[k]) {
                    indexes[i - 65] = min(indexes[i - 65], k);
                    break;
                }
            }
        }
    }
    
    for(int i = 0; i < targets.size(); i++) {
        for(auto& t : targets[i]) {
            if(indexes[t - 65] == 101) {
                answer[i] = -1;
                break;
            }
            answer[i] += indexes[t - 65] + 1;
        }
    }
    
    return answer;
}